feat: add container to multiple text elements

ChinmayMhatre opened the pull request:
Closes #6418

Updated `actionCreateContainerFromText` action to take multiple text elements.
Iterated over each element and adding containers to them


https://user-images.githubusercontent.com/51131670/230577307-4a6a56af-94c1-4ec5-ac6a-9fa515fea2f7.mov

dwelle commented on src/actions/actionBoundText.tsx:
```diff
@@ -185,107 +185,111 @@ export const actionCreateContainerFromText = register({
-          }
+    let updatedElements = elements.slice();
+    let updatedAppState = appState;
+    let containerIds:any = {};
```
let's use the `AppState["selectedElementIds"]` type here

dwelle commented:
Looks good @ChinmayMhatre!

In the original issue I suggested spreading the elements as well to ensure they don't overlap after containerizing, but it may not be so simple. I'll create a new issue for it and we can do it separately.

ChinmayMhatre commented:
> Looks good @ChinmayMhatre!
> 
> In the original issue I suggested spreading the elements as well to ensure they don't overlap after containerizing, but it may not be so simple. I'll create a new issue for it and we can do it separately.

Yes, it could come in another issue. I gave it a try but couldn't get it to work so I thought of pushing just this commit first.

ChinmayMhatre commented:
> Looks good @ChinmayMhatre!
> 
> In the original issue I suggested spreading the elements as well to ensure they don't overlap after containerizing, but it may not be so simple. I'll create a new issue for it and we can do it separately.

Yes we could do that ! I have updated the type and fixed some linting warnings. Let me know if anything else is needed!

dwelle commented:
Thanks for the PR, we'll ship soon ❤️

Fran5ois commented:
Thank you very much for implementing this!

There seems to be a limitation: 

When one of the text elements selected is already wrapped in a container, the action doesn't show anymore in the context menu...

I opened an [issue ](https://github.com/excalidraw/excalidraw/issues/9334) for this...
